WordPress/gutenberg
# [RNMobile] Search Block: Settings

## Description
This PR builds on the branch for #29096 and as such, that PR must be merged before this one can be taken out of draft. This PR is a continuation in the saga to port the Search Block to mobile. It adds the block settings component along with the `width` setting:
- Units supported: percentage and pixel
- Defaults and minimums included.  

**Note included:**
- There will be a project in the future for moving the block toolbar options to settings to make it more accessible for mobile. That change is not included in this PR. 

## How has this been tested?



Tested using the demo app and WordPress apps on an Android emulator (android 11) and iOS simulator (iOS 14.4). 

1. Add a **Search Block** to the editor.
2. Open the **block settings**, verify **Width** setting is displayed and that it defaults to **100%**. 
3. Change the **width** to **70% by dragging the slider**. **Manually** enter in a value of **75%**. Close settings and open HTML Editor - verify block setting changed. 
4. Open the **block settings** again, change the unit to **pixels**. Verify slider disappears and is replaced by a stepper. Default width should be **350 px**. 
5. Click the **+** and **-** buttons to verify width changes as expected. 
6. Change the **width to 500 pixels** by **manually** entering the value. Close settings and verify change in HTML editor. 
7. Try setting the **width to 888%**, the width should default to he **max of 100%**.
8. Try setting the **width to 100 px**, the width should default to the **minimum width of 220 px**.

## Screenshots 

https://user-images.githubusercontent.com/5810477/108923391-b72d1b00-7606-11eb-8951-3eb35b4a3c9d.mov


## Types of changes




New feature - part of a larger project that will include multiple PRs. 

## Checklist:
- [x] I've tested my changes with keyboard and screen readers.  (Will be handled in a later PR)
